backend/fms_core/template_prefiller/prefiller.py

```diff
@@ -19,10 +19,15 @@ def PrefillTemplate(template_path, template_info, queryset):
 
     workbook = load_workbook(filename=template_path)
     position_dict = load_position_dict(workbook, template_info["sheets info"], template_info["prefill info"])
-
     for sheet_name, sheet_dict in position_dict.items():
         current_sheet = workbook[sheet_name]
-        for i, entry in enumerate(queryset.values(*filter(lambda x: x is not None, sheet_dict["queryset_column_list"]))):
+        is_sheet_pure_batch = is_sheet_true_batch(sheet_name, template_info["sheets info"])
+        if is_sheet_pure_batch:
+            queryset = queryset.values(*filter(lambda x: x is not None, sheet_dict["queryset_column_list"]))
+            queryset = queryset.order_by().distinct()
+        else:
+            queryset = queryset.values(*filter(lambda x: x is not None, sheet_dict["queryset_column_list"]))
```

Comment:
Why not use list comprehension style filter? ;)

Reply:
Duplication could be reduced by moving `queryset = queryset.values(*filter(lambda x: x is not None, sheet_dict["queryset_column_list"]))` on top of the `if` statement

Reply:
as for comprehension, i just kept the existing code... I refactor as little as i can ...